sonar-html-plugin/src/test/java/org/sonar/plugins/html/lex/PageLexerTest.java

```diff
@@ -503,8 +502,34 @@ public void cdata() {
   }
 
   @Test
-  public void tag_with_whitespace_before_name() {
-    assertSingleTag("<  html>");
+  public void tag_with_whitespace_after_name() {
+    assertSingleTag("<html  >");
+  }
+
+  @Test
+  public void tag_with_invalid_character_before_name_is_considered_as_text() {
+    // Tag names cannot start with a whitespace, a digit or any other invalid character: https://www.w3.org/TR/REC-xml/#sec-starttags.
+    assertOnlyText("<  html>");
+    assertOnlyText("<5html>");
+    assertOnlyText("<\u2190html>");
+  }
+
+  @Test
+  public void start_tag_character_is_considered_as_text_when_followed_by_whitespace() {
+    PageLexer lexer = new PageLexer();
+    List<Node> nodes = lexer.parse(new StringReader("<a> < a </a>"));
+    assertThat(nodes).hasSize(4);
+    assertThat(nodes).extracting(Node::getNodeType).containsExactly(NodeType.TAG, NodeType.TEXT, NodeType.TEXT, NodeType.TAG);
+    assertThat(nodes).extracting(Node::getCode).containsExactly("<a>", " ", "< a ", "</a>");
```

Comment:
_curiosity_ Do you see any potential issue having two or more consecutive Text nodes? Shouldn't we merge them ?

Reply:
I thought about it, but merging them was not so simple to do easily.
I've checked, and the only checks using text nodes are `LongJavaScriptCheck`, `FileLengthCheck`, `LinksIdenticalTextsDifferentTargetsCheck`, `ObjectWithAlternativeContentCheck`, and `InternationalizationCheck`. For the first four, having consecutive text nodes is not a problem, as the text is either appended in the rule, or the check itself makes it so it does not matter.
For `InternationalizationCheck`, it could potentially lead to a FP in the case there is a '<' character in a text node and the characters before it are only letters or digits. Ex: `<span>r< </span>`